tensorflow/lite/micro/tools/generate_test_for_model.py

```diff
@@ -0,0 +1,207 @@
+                          high=np.iinfo(dtype).max,
+                          dtype=dtype,
+                          size=input_tensor().shape),
+    ]
+
+  def generate_inputs_add_sub(self, interpreter, dtype):
+    input_tensor0 = interpreter.tensor(
+        interpreter.get_input_details()[0]['index'])
+    input_tensor1 = interpreter.tensor(
+        interpreter.get_input_details()[1]['index'])
+    return [
+        np.random.randint(low=np.iinfo(dtype).min,
+                          high=np.iinfo(dtype).max,
+                          dtype=dtype,
+                          size=input_tensor0().shape),
+        np.random.randint(low=np.iinfo(dtype).min,
+                          high=np.iinfo(dtype).max,
+                          dtype=dtype,
+                          size=input_tensor1().shape)
+    ]
+
+  def generate_inputs_transpose_conv(self, interpreter, dtype):
+    input_tensor0 = interpreter.tensor(0)
+    filter_tensor = interpreter.tensor(1)
+    input_tensor1 = interpreter.tensor(2)
+    output_shape = np.array([
+        1,
+        filter_tensor().shape[1] + input_tensor1().shape[1] - 1,
+        int(input_tensor1().shape[2] * 2 + filter_tensor().shape[2] / 2),
+        filter_tensor().shape[0]
+    ],
+                            dtype=np.int32)
+    return [
+        output_shape,
+        np.random.randint(low=np.iinfo(dtype).min,
+                          high=np.iinfo(dtype).max,
+                          dtype=dtype,
+                          size=input_tensor1().shape)
+    ]
+
+  def GetTypeStringFromTensor(self, tensor):
+    if tensor.dtype == np.int8:
+      return 'int8'
+    if tensor.dtype == np.int16:
+      return 'int16'
+    if tensor.dtype == np.int32:
+      return 'int32'
+    if tensor.dtype == np.float:
+      return 'float'
+
+  def generate_golden(self):
+    model_path = self.model_paths[0]
+    interpreter = tf.lite.Interpreter(model_path=model_path)
+
+    interpreter.allocate_tensors()
+
+    input_details = interpreter.get_input_details()
+    if len(input_details) > 1:
+      raise RuntimeError(f'Only models with one input supported')
+    input_tensor = interpreter.tensor(
+        interpreter.get_input_details()[0]['index'])
+    output_tensor = interpreter.tensor(
+        interpreter.get_output_details()[0]['index'])
+
+    input_type = interpreter.get_input_details()[0]['dtype']
+    output_type = interpreter.get_output_details()[0]['dtype']
+    if input_type != np.int8 or output_type != np.int8:
+      raise RuntimeError(f'Only int8 models supported')
+
+    generated_inputs = self.generate_inputs_single(interpreter,
+                                                   input_tensor().dtype)
+    for i, _input_detail in enumerate(input_details):
+      interpreter.set_tensor(input_details[i]["index"], generated_inputs[i])
+
+    interpreter.invoke()
+
+    self.write_golden(generated_inputs, model_path, output_tensor)
+
+  def generate_goldens(self, builtin_operator):
```

Comment:
I added some documentation but I missed your second question while doing this. Do you think this doc is enough?
I am not sure if it has to be only one operator but that is the use case for TestModelGenerator that is using this and it is kind of implied since builtin_operator argument is not a list and can only contain one builtin_operator.

Reply:
This function figures out how to generate input based on builtin_operator; so the very first operator in the graph has to be of type of builtin_operator. 
Ideally, I wish generate_goldens does NOT need to have the builtin_operator argument because by using flatbuffer api because the function can look at the first OP and figure out its first operator and that would make it more generic. But before that, since the function is now in the tools directory, and not tied to TestModelGenerator, I feel a user could be confused by where this builitin_operator come from and what for; hence I think more documentation about builtin_operator is helpful in reducing the reader's cognitive load

Reply:
Done